apps/ledger-live-desktop/src/renderer/families/evm/EditTransaction/Body.tsx

```diff
@@ -0,0 +1,336 @@
+  const [parentAccount, setParentAccount] = useState<Account | null | undefined>(
+    parentAccountFromBridge,
+  );
+
+  invariant(account, "account required");
+  invariant(transactionToUpdate, "transactionToUpdate required");
+  invariant(transaction, "original transaction required");
+
+  const currencyName = getAccountCurrency(account).name;
+
+  const handleCloseModal = useCallback(() => {
+    closeModal("MODAL_EVM_EDIT_TRANSACTION");
+  }, [closeModal]);
+
+  const handleRetry = useCallback(() => {
+    setTransactionError(null);
+    setOptimisticOperation(null);
+    setSigned(false);
+  }, []);
+
+  const handleTransactionError = useCallback((error: Error) => {
+    if (!(error instanceof UserRefusedOnDevice)) {
+      logger.critical(error);
+    }
+    setTransactionError(error);
+  }, []);
+
+  const handleOperationBroadcasted = useCallback(
+    (optimisticOperation: Operation) => {
+      const mainAccount = getMainAccount(account, parentAccount);
+      updateAccountWithUpdater(mainAccount.id, account =>
+        addPendingOperation(account, optimisticOperation),
+      );
+      setOptimisticOperation(optimisticOperation);
+      setTransactionError(null);
+    },
+    [account, parentAccount, updateAccountWithUpdater],
+  );
+
+  const handleStepChange = useCallback(step => onChangeStepId(step.id), [onChangeStepId]);
+  const error = transactionError || bridgeError;
+  const mainAccount = getMainAccount(account, parentAccount);
+
+  useEffect(() => {
+    const setTransactionHasBeenValidatedCallback = async () => {
+      const tx = await getTransactionByHash(mainAccount.currency, params.transactionHash);
+      if (tx?.confirmations) {
+        // stop polling as soon as we have a confirmation
+        clearInterval(intervalId);
+        setTransactionHasBeenValidated(true);
+      }
+    };
+
+    setTransactionHasBeenValidatedCallback();
+    const intervalId = setInterval(
+      () => setTransactionHasBeenValidatedCallback(),
+      mainAccount.currency.blockAvgTime
+        ? mainAccount.currency.blockAvgTime * 1000
+        : getEnv("DEFAULT_TRANSACTION_POLLING_INTERVAL"),
+    );
+
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, [mainAccount.currency, params.transactionHash]);
+
+  const haveFundToCancel = hasMinimumFundsToCancel({ transactionToUpdate, mainAccount });
+  const haveFundToSpeedup = hasMinimumFundsToSpeedUp({ transactionToUpdate, mainAccount, account });
+
+  // if we are at this step (i.e: in this screen) it means the transaction is editable
+  const isOldestEditableOperation = isOldestPendingOperation(
+    mainAccount,
+    transactionToUpdate.nonce,
+  );
+
+  const [editType, setEditType] = useState<StepProps["editType"]>(
+    isOldestEditableOperation && haveFundToSpeedup
+      ? "speedup"
+      : haveFundToCancel
+      ? "cancel"
+      : undefined,
+  );
+  const handleSetEditType = useCallback(editType => setEditType(editType), []);
```

Comment:
Not 100% but if the array is empty then it's just the same as if there was no useCallback at all

```suggestion
  const handleSetEditType = setEditType(editType);
```
Both are the same

Reply:
> Not 100% but if the array is empty then it's just the same as if there was no useCallback at all

Not exactly, `useCallback` is used to return a cached version of a function (the first param of `useCallback`). And the second param of `useCallback` is the dependency array, used to determine when to invalidate this cache.

Since in JavaScript, a function () {} or () => {} always creates a different function, without `useCallback`, the `handleSetEditType` function will always be a different reference, at each render, even though the function "did not change" (i.e: still does the same thing).

But then a question could be "why do we need to keep the same reference between renders for this function?"

As stated by the [`useCallback` doc](https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere):
> Caching a function with useCallback  is only valuable in a few cases:
> - ...
> - The function you’re passing is later used as a dependency of some Hook. For example, another function wrapped in useCallback depends on it, or you depend on this function from [useEffect.](https://react.dev/reference/react/useEffect)

Here, we are in this case where the `handleSetEditType` function, passed as the `setEditType` prop of the `Stepper` component, is latter used as a dependency of a call to `useCallback` in the `StepMethod` component: 

https://github.com/LedgerHQ/ledger-live/blob/cc7761e190aa0717e7c55c269e2f035b65ba5c95/apps/ledger-live-desktop/src/renderer/families/evm/EditTransaction/steps/StepMethod.tsx#L80-L90

Here we use `useCallback` in order to avoid unnecessary function re-creation, which could lead to unnecessary re-renders.